src/server/replica.cc

```diff
@@ -145,6 +145,11 @@ void Replica::Stop() {
     state_mask_.store(0);  // Specifically ~R_ENABLED.
   });
 
+  CloseSocket();
+  for (auto& flow : shard_flows_) {
+    flow->Cancel();
+  }
+
```

Comment:
So why doesn't the error handler of the context do this?

Reply:
because it's not error case, we need to stop replica and we need close the sockets to prevent waiting data

Reply:
Why isn't it an error case? cntx->Cancel() above issues a errc::operaction_canceled error

Reply:
yes we issue an error, but how before this cancel we can stop the connections

Reply:
We do this

`cntx_.SwitchErrorHandler([this](const auto& ge) { this->DefaultErrorHandler(ge); });`

which should close all connections on an error (which cancellation is)

Reply:
we need the firtst error to happen and I've checked that it works

Reply:
I am not quite sure I understand this change either. So for `Stable Sync` the entry point of the flow is:

```
 625 error_code Replica::ConsumeDflyStream() {                                                           
 626   // Set new error handler that closes flow sockets.                                                
 627   auto err_handler = [this](const auto& ge) {                                                       
 628     // Trigger acl-checker                                                                          
 629     replica_waker_.notifyAll();                                                                     
 630     // Make sure the flows are not in a state transition                                            
 631     lock_guard lk{flows_op_mu_};                                                                    
 632     DefaultErrorHandler(ge);                                                                        
 633     for (auto& flow : shard_flows_) {                                                               
 634       flow->Cancel();                                                                               
 635     }                                                                                               
 636     multi_shard_exe_->CancelAllBlockingEntities();                                                  
 637   };                                                                                                
 638   RETURN_ON_ERR(cntx_.SwitchErrorHandler(std::move(err_handler)));     ****
```

Note that `DefaultErrorHandler` internally calls `CloseSocket` and `DflyShardReplica::Cancel` also does the same.

So when we call `cntx_.Cancel()` on line 145 it triggers the the error handler registered above:

```
 627   auto err_handler = [this](const auto& ge) {
 628     // Trigger acl-checker
 629     replica_waker_.notifyAll();                                                                     
 630     // Make sure the flows are not in a state transition                                            
 631     lock_guard lk{flows_op_mu_};                                                                    
 632     DefaultErrorHandler(ge);                                                                        
 633     for (auto& flow : shard_flows_) {
 634       flow->Cancel();
 635     }                                                                                               
 636     multi_shard_exe_->CancelAllBlockingEntities();                                                  
 637   };
```

Which 
1. Calls `CloseSocket` via `DefaultErrorHandler` 
2. Calls `flow->Cancel()` which closes each of the flows socket respectively

I am really unsure how this change affects anything

Reply:
Thx I got your point. I will investigate this issue deeper.